custom/samveg/case_importer/validators.py

```diff
@@ -38,6 +38,32 @@ def _validate_missing_columns(cls, spreadsheet):
             )]
 
 
+class MandatoryValueValidator(BaseValidator):
+    @classmethod
+    def run(cls, row_num, raw_row, fields_to_update):
+        error_messages = []
+        error_messages.extend(cls._validate_mandatory_columns(row_num, raw_row))
+        return fields_to_update, error_messages
+
+    @classmethod
+    def _validate_mandatory_columns(cls, row_num, raw_row):
+        error_messages = []
+        missing_values = set()
+
+        columns = set(raw_row.keys())
+        mandatory_columns = get_mandatory_columns(columns)
+
+        for mandatory_column in mandatory_columns:
```

Comment:
Ya, I did use set earlier but since the order in set could change, the UI would have 2 different error messages for the same set of missing fields. Like if 2 rows have missing A and B column, they might end up with messages as "Missing values for A, B" and "Missing values for B, A", leading to two separate entries in UI.
So to have predictable order I had to use lists.